R/learner_partykit_surv_cforest.R

```diff
@@ -133,21 +133,20 @@ LearnerSurvCForest = R6Class("LearnerSurvCForest",
       preds = invoke(predict, object = self$model, newdata = newdata,
         type = "prob", .args = pars)
 
-      # Define WeightedDiscrete distr6 distribution from the survival function
-      x = lapply(preds, function(z) {
-        time = c(0, z$time, max(z$time) + 1e-3)
-        surv = c(1, z$surv, 0)
-        data.frame(x = time, cdf = 1 - surv)
-      })
-      distr = distr6::VectorDistribution$new(
-        distribution = "WeightedDiscrete",
-        params       = x,
-        decorators   = c("CoreStatistics", "ExoticStatistics"))
+      times = lapply(preds, function(p) p$time)
+      utimes = sort(unique(unlist(times)))
 
-      # Define crank as the mean of the survival distribution
-      crank = -vapply(x, function(z) sum(z[, 1] * c(z[, 2][1], diff(z[, 2]))), numeric(1))
+      # to use non-exported function from `distr6`
+      extend_times = getFromNamespace("C_Vec_WeightedDiscreteCdf", ns = "distr6")
+      res = lapply(preds, function(p) {
+        # p is a `survfit` object
+        cdf = matrix(data = 1 - p$surv, ncol = 1) # 1 observation (column), rows => times
+        # extend cdf to 'utimes', return survival
+        extend_times(utimes, p$time, cdf = cdf, FALSE, FALSE)
+      })
+      surv = do.call(cbind, res) # rows => times, columns => obs
 
-      list(crank = crank, distr = distr)
+      .surv_return(times = utimes, surv = t(surv))
```

Comment:
if you now transpose surv, why don't you create it the right way in the first place?

Reply:
It's the output of the `C_Vec_WeightedDiscreteCdf` (`extend_times` now) that has the rows and columns reversed and in the for loop above you get each row of the end matrix reversed via this function. Which means that you either do a `t` transpose on each observation (making a column to a row) after calling `extend_times`, or what I did (leave all columns, `cbind` them, transpose at the end) => second is faster was my thought

Reply:
makes sense!